### PR TITLE
Paygas Refactoring

### PR DESCRIPTION
**Pull request by adietrichs**
Moves the `PAYGAS` related settings out of `vmConfig` into `Context`.

### CODE REVIEW

File: core/evm.go [hunk 1]
```diff
@@ -44,6 +44,14 @@ func NewEVMContext(msg Message, header *types.Header, chain ChainContext, author
 	} else {
 		beneficiary = *author
 	}
+	var paygasMode = vm.PaygasNoOp
+	if msg.IsAA() {
+		if msg.To() != nil {
```
Comment (villanuevawill) [inline core/evm.go]: hmmm, why does msg.To() == nil turn on PaygasHalt mode? this seems generally hacky.

**Review by villanuevawill — CHANGES_REQUESTED** [review]
Suggested comment

File: core/evm.go (hunk 1 above)
Comment (adietrichs) [reply core/evm.go]: Good point. I removed the distinction between `PaygasContinue` and `PaygasHalt` from `NewEVMContext`.
I now always set it to `PaygasContinue` there and then to `PaygasHalt` during `Validate()`

**Review by villanuevawill — APPROVED** [review]
LGTM

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**